tests/std/include/parallel_algorithms_utilities.hpp

```diff
@@ -7,9 +7,10 @@
 #include <thread>
 
 #ifdef EXHAUSTIVE
-const size_t max_parallel_test_case_n = 1000;
+const size_t max_parallel_test_case_n = 4096;
 #else // ^^^ EXHAUSTIVE ^^^ // vvv !EXHAUSTIVE vvv
-const size_t max_parallel_test_case_n = std::max(4u, std::thread::hardware_concurrency()) + 1;
+// The constant 32 comes from std::_Oversubscription_multiplier
+const size_t max_parallel_test_case_n = std::max(4u, std::thread::hardware_concurrency() * 32) + 1;

```

Comment:
We observed `P0024R2_parallel_algorithms_set_difference` and `P0024R2_parallel_algorithms_set_intersection` timeouts internally; I suspect that this is involved.

https://github.com/microsoft/STL/blob/ef6c1ce124cb99115825bfdbf4ed6018ebe8a301/tests/std/include/parallel_algorithms_utilities.hpp#L17-L29

https://github.com/microsoft/STL/blob/ef6c1ce124cb99115825bfdbf4ed6018ebe8a301/tests/std/tests/P0024R2_parallel_algorithms_set_difference/test.cpp#L183

https://github.com/microsoft/STL/blob/ef6c1ce124cb99115825bfdbf4ed6018ebe8a301/tests/std/tests/P0024R2_parallel_algorithms_set_difference/test.cpp#L30-L42

So `testSize` can be 32N. `shortList` contains `testSize / 2` elements, so 16N. Then we iterate for every `overlappingPoint` in `shortList`:

https://github.com/microsoft/STL/blob/ef6c1ce124cb99115825bfdbf4ed6018ebe8a301/tests/std/tests/P0024R2_parallel_algorithms_set_difference/test.cpp#L98-L101

So we're processing 16N elements, 16N times. That's 256 N^2, where N is `thread::hardware_concurrency()`; for N=64 this is 1M.

The tests don't seem to take *too* long on my 32-core machine (256K), but I do notice that they take a while (28 seconds), and my physical machine is very fast. I suspect that with 4x intensity on the slower VMs, we could easily be exceeding the timeouts we set internally (6 minutes), given that each test tries to consume all cores, so if a bunch of other tests are running, very few cores may be effectively available. (In fact, the N^2 pressure means that this gets worse for larger machines; on average each test will have 1 core available because the test runner is already trying to run N tests simultaneously.)

Although we haven't decisively identified the cause of these timeouts, I suspect that something needs to change in this PR.